### PR TITLE
Add PointCloud2 support for RangeFinder.

### PR DESCRIPTION
**Pull request by TaoYibo1866**
**Description**
Add PointCloud2 support for RangeFinder.

**Related Issues**
This pull-request fixes issue #227

**Affected Packages**
List of affected packages:
  - webots_ros2_driver

### CODE REVIEW

**Comment by BenjaminHug8** [comment]
@TaoYibo1866 Thank you for your contribution, I just made slightly modification to respect the convention of ROS 2 msgs and to organize the code, but it was already a nice job!

@omichel May I ask you to quickly check this and confirm everything is ok?

File: webots_ros2_driver/src/plugins/static/Ros2RangeFinder.cpp
Comment (omichel) [inline webots_ros2_driver/src/plugins/static/Ros2RangeFinder.cpp:55]: I would rename these variables:
```suggestion
    const double focalLengthX = 0.5 * width * (1 / tan(0.5 * mRangeFinder->getFov()));
    const double focalLengthY = 0.5 * height * (1 / tan(0.5 * mRangeFinder->getFov()));
```

**Review by omichel — CHANGES_REQUESTED** [review]
It looks pretty good to me.
Thank you.

**Review by omichel — DISMISSED** [review]
Thank you both.

**Comment by TaoYibo1866** [comment]
@BenjaminHug8 Thank you for your work, but i am confused about the size of a single point. https://github.com/cyberbotics/webots_ros2/blob/3f8b3e78fefb381299b1a9a1c21a36ef3f3a8687/webots_ros2_driver/src/plugins/static/Ros2RangeFinder.cpp#L87-L89
I think the size of one point is 3 * sizeof(float) = 12 instead of 4. Do i misunderstand the concept?

**Comment by BenjaminHug8** [comment]
> I think the size of one point is 3 * sizeof(float) = 12 instead of 4. Do i misunderstand the concept?

No no you are right, it is my mistake. I was thinking about a 2D image where the point is just a depth, but that is not the case here (jumping from PRs to PRs).

Thank you!

**Review by BenjaminHug8 — APPROVED** [review]
Thank you @TaoYibo1866, I let you the honor to press the squash and merge button!

**Comment by TaoYibo1866** [comment]
> Thank you @TaoYibo1866, I let you the honor to press the squash and merge button!

That is very kind of you, i am flattered.
But I have never done this before, do you mean press the red "Close pull request" button?

**Comment by BenjaminHug8** [comment]
> do you mean press the red "Close pull request" button?

@omichel will give you the permission to do it.

**Comment by omichel** [comment]
Hi @TaoYibo1866, I just sent you a GitHub invite to become a member of the Webots Committers group, so that you should be able to merge this PR after you accept my invite.

**Comment by TaoYibo1866** [comment]
I am really flattered. It's my honor to join in this amazing project. I will "Squash and merge" now for the first time, thank you for giving me this chance!

**Comment by omichel** [comment]
You are welcome! Thank you for your contribution.

**Pull request merged**
